app/javascript/script.js

```diff
@@ -0,0 +1,121 @@
+document.addEventListener("DOMContentLoaded", function () {
+    const rangeInput = document.querySelectorAll(".range-input input"),
+    priceInput = document.querySelectorAll(".price-input input"),
+    range = document.querySelector(".slider .progress");
+    let priceGap = 1;
+    
+    // Function to get URL query parameters
+    function getQueryParams(url) {
+        const params = {};
+        const searchParams = new URLSearchParams(url.split("?")[1]);
+        for (let param of searchParams.entries()) {
+        params[param[0]] = param[1];
+        }
+        return params;
+    }
+
+    // Function to update the URL query parameters with selected price range
+    function updateURLParams(minPrice, maxPrice) {
```

Comment:
this is not wrong, but why not simply submit a form from the HTML and avoid all this extra work?